ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,255 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Shapes 1.15
+
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+import shared.stores.send 1.0
+import shared.popups.send.views 1.0
+import shared.popups.send.panels 1.0
+
+import utils 1.0
+
+Control {
+    id: root
+
+    // input API
+    required property TransactionStore store // TODO refactor to accept SwapModalAdaptor/SwapStore ???
+
+    property string inputTokenKey
+    onInputTokenKeyChanged: Qt.callLater(d.setSelectedHoldingId, inputTokenKey, Constants.TokenType.ERC20)
+    property string inputTokenAmount
+    onInputTokenAmountChanged: Qt.callLater(() => amountToSendInput.input.text =
+                                            Number(inputTokenAmount).toLocaleString(Qt.locale(), 'f', -128))
+
+    property int swapSide: SwapInputPanel.SwapSide.Pay
+
+    // output API
+    readonly property string selectedHoldingId: d.selectedHoldingId
+    readonly property double cryptoValue: amountToSendInput.cryptoValueToSendFloat
+    readonly property string cryptoValueRaw: amountToSendInput.cryptoValueToSend
+    readonly property bool cryptoValueValid: amountToSendInput.inputNumberValid
+
+    // visual properties
+    property int swapExchangeButtonWidth: 44
+    property string caption: swapSide === SwapInputPanel.SwapSide.Pay ? qsTr("Pay") : qsTr("Receive")
+
+    enum SwapSide {
+        Pay = 0,
+        Receive = 1
+    }
+
+    padding: Style.current.padding
+
+    // by design
+    implicitWidth: 492
+    implicitHeight: 131
+
+    QtObject {
+        id: d
+
+        function setSelectedHoldingId(holdingId, holdingType) {
```

Comment:
can we find a way to simplify all of this? I find the flow to get the holding from store a bit weird.  can we not just be using keys to identify whats being used?

Reply:
Yeah I'll think of something; the store acts as a helper here only

Reply:
I'd leave that for the HoldingSelector refactoring